dome9/resource_dome9_organizational_unit_test.go

```diff
@@ -0,0 +1,112 @@
+package dome9
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/dome9/dome9-sdk-go/services/organizationalunits"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/resourcetype"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/method"
+	"github.com/terraform-providers/terraform-provider-dome9/dome9/common/testing/variable"
+)
+
+func TestAccResourceOrganizationalUnitBasic(t *testing.T) {
+	var ouResponse organizationalunits.OUResponse
+	resourceTypeAndName, _, generatedName := method.GenerateRandomSourcesTypeAndName(resourcetype.OrganizationalUnit)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckOrganizationalUnitDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckOrganizationalUnitConfigure(resourceTypeAndName, generatedName, variable.OrganizationalUnitName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOrganizationalUnitExists(resourceTypeAndName, &ouResponse),
+					resource.TestCheckResourceAttr(resourceTypeAndName, "name", variable.OrganizationalUnitName),
+					resource.TestCheckResourceAttr(resourceTypeAndName, "parent_id", variable.ParentID),
+				),
+			},
+
+			// Update test
+			{
+				Config: testAccCheckOrganizationalUnitConfigure(resourceTypeAndName, generatedName, variable.OrganizationalUnitNameUpdate),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOrganizationalUnitExists(resourceTypeAndName, &ouResponse),
+					resource.TestCheckResourceAttr(resourceTypeAndName, "name", variable.OrganizationalUnitNameUpdate),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckOrganizationalUnitDestroy(s *terraform.State) error {
+	apiClient := testAccProvider.Meta().(*Client)
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != resourcetype.OrganizationalUnit {
+			continue
+		}
+
+		ipList, _, err := apiClient.ruleSet.Get(rs.Primary.ID)
```

Comment:
replace ruleSet with Organizationalunit and change the name of the variable ipList